tensorflow/core/grappler/optimizers/implementation_selector.cc

```diff
@@ -261,12 +272,37 @@ Status ImplementationSelector::MaybeOptimizeFunctionCall(
   }
 
   DeviceNameUtils::ParsedName parsed_name;
-  if (!DeviceNameUtils::ParseFullName(node_def->device(), &parsed_name) ||
-      !parsed_name.has_type) {
-    return errors::Internal("Could not parse device name:", node_def->device());
+  if (!node_def->device().empty()) {
+    if (!DeviceNameUtils::ParseFullName(node_def->device(), &parsed_name) ||
+        !parsed_name.has_type) {
+      return errors::Internal("Could not parse device name:",
```

Comment:
We are using absl for error reporting in new code. Please use `absl::InternalError` instead. Here is an example usage:
https://github.com/tensorflow/tensorflow/blob/f79cfa4e6de05c38a589cdbd3560da42ad1af8dd/tensorflow/core/kernels/fft_ops.cc#L731

Reply:
Fixed.